scripts/run.sh

```diff
@@ -8,31 +19,142 @@ function depends {
   fi
 }
 
-if [ ! -d ".git" ]; then 
-  echo "Please run this script in your top level directory."
-  exit 1
-fi
+is_email_valid() {
+  regex="^([A-Za-z]+[A-Za-z0-9]*((\.|\-|\_)?[A-Za-z]+[A-Za-z0-9]*){1,})@(([A-Za-z]+[A-Za-z0-9]*)+((\.|\-|\_)?([A-Za-z]+[A-Za-z0-9]*)+){1,})+\.([A-Za-z]{2,})+"
+  [[ "${1}" =~ $regex ]]
+}
+
+
+check_depend() {
+  depends docker
+  depends docker-compose
+}
+
+check_directory() {
+  if [ ! -d ".git" ]; then 
+    echo "Error: Please run this script in your top level directory."
+    exit 1
+  fi
+}
+
+run_dev() {
+  if [ ! -f "docker-compose.yml" ]; then 
+    echo "Error: docker-compose.yml missing, which is required for docker-compose"
+    echo "Please check the repo for missing files"
+    exit 1
+  fi
+
+  export SDSCHEDULE_SCRAPE
+
+  docker-compose up ${is_build}
+}
+
+run_prod() {
+  if [ ! -f "docker-compose-production.yml" ]; then 
+    echo "Error: docker-compose-production.yml missing, which is required for docker-compose"
+    echo "Please check the repo for missing files"
+    exit 1
+  fi
+
+  export SDSCHEDULE_SCRAPE
+
+  echo "Starting up production servers"
 
-download=false
+  docker-compose -f docker-compose-production.yml up ${is_build} --detach
+}
+
+stop_prod() {
+  if [ ! -f "docker-compose-production.yml" ]; then 
+    echo "Error: docker-compose-production.yml missing, which is required for docker-compose"
+    echo "Please check the repo for missing files"
+    exit 1
+  fi
+
+  echo "Stopping containers in production mode"
+
+  docker-compose -f docker-compose-production.yml down
+}
+
+run_certbot() {
+  echo
+  echo "Will generate a brand new cert"
+  echo
+  docker exec sdschedule-certbot sed -i -r 's/(listen .*443)/\1;#/g; s/(ssl_(certificate|certificate_key|trusted_certificate) )/#;#\1/g' /etc/nginx/sites-enabled/sdschedule.conf
+  docker exec sdschedule-web nginx -s reload
+  docker exec sdschedule-certbot certbot certonly --webroot -w /var/www/certbot -d sdschedule.com -n --force-renewal --agree-tos --email "$letsencrypt_email"
+  docker exec sdschedule-certbot sed -i -r 's/#?;#//g' /etc/nginx/sites-enabled/sdschedule.conf
+  docker exec sdschedule-web nginx -s reload
+}
+
+main() {
+  check_directory
+  check_depend
+
+  case $environment in
+    PROD)
+      run_prod
+    ;;
+    DEV)
+      run_dev
+    ;;
+    *)
+      echo "Internal Error: Invalid Environment Setting"
+      exit -1
+    ;;
+  esac
+}
+
+if [ $# -eq 0 ]; then
+  echo "Run in development mode without downloading and rebuilding by default"
+  echo "Try '-h' or '--help' for help message"
+  echo
```

Comment:
If there are no parameters you should exit here

Reply:
If exit, then the user has no way to run the app in dev mode without downloading or rebuilding.

Reply:
Alright that sounds fine